ext/openssl/openssl.c

```diff
@@ -6801,32 +6802,34 @@ PHP_FUNCTION(openssl_random_pseudo_bytes)
 	zend_string *buffer = NULL;
 	zval *zstrong_result_returned = NULL;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l|z/", &buffer_length, &zstrong_result_returned) == FAILURE) {
-		return;
-	}
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_LONG(buffer_length)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_ZVAL_DEREF(zstrong_result_returned)
+	ZEND_PARSE_PARAMETERS_END();
 
 	if (zstrong_result_returned) {
 		zval_ptr_dtor(zstrong_result_returned);
-		ZVAL_FALSE(zstrong_result_returned);
+		ZVAL_TRUE(zstrong_result_returned);
+		php_error_docref(NULL, E_DEPRECATED, "Calling openssl_random_pseudo_bytes() with the second argument is deprecated");
 	}
 
 	if (buffer_length <= 0
 #ifndef PHP_WIN32
 		|| ZEND_LONG_INT_OVFL(buffer_length)
 #endif
 			) {
-		RETURN_FALSE;
+		zend_throw_exception(zend_ce_error, "Length must be greater than 0", 0);
```

Comment:
I think this can potentially break stuff. I'm fine with throwing on rand error which happens only if there is something really wrong with the environment but this case can happen more likely. Maybe we should reconsider this case and just return empty string or keep it as `false`.

Reply:
Thanks for the review @bukka! This is to mimic the behavior of `random_bytes()` like the RFC suggests. This will ensure that `openssl_random_pseudo_bytes()` can't be used as an attack vector in the event an attacker has access to change `$length`.

```php
// Let's hope this doesn't exist in the wild! :)
$bytes = openssl_random_pseudo_bytes($_GET['length']);
```

Reply:
I'm also somewhat concerned about this one. Requesting zero bytes from an RNG is not wrong per-se -- it's an operation with a well-defined return value, namely the empty string. The current return value (false) happens to coerce to that.

I feel like there's not much value in throwing an exception for this one specific case (throwing for negative length or overflowing length is fine), and I strongly suspect that this will lead to BC breakage somewhere, particularly in testing code. E.g. I remember that when we merged in ext/sodium and switched from using a sodium-specific random bytes implementation (which allows 0 length) to PHP's that's one of the issues that came up in testing code that was testing random strings of random length (including zero).

That's why I would prefer to continue allowing this particular case. As the RFC does not spell this out either way, I think we could still do so, if we wanted :)